controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -403,19 +405,26 @@ public void listStreamsInScope(Controller.StreamsInScopeRequest request, StreamO
         authenticateExecuteAndProcessResults(
                 () -> this.authHelper.checkAuthorization(AuthResourceRepresentation.ofScope(scopeName),
                         AuthHandler.Permissions.READ),
-                delegationToken -> controllerService.listStreams(
-                        scopeName, request.getContinuationToken().getToken(), listStreamsInScopeLimit)
-                        .thenApply(response -> {
-                             List<StreamInfo> streams = response.getKey().stream()
-                                     .filter(streamName -> authHelper.isAuthorized(
-                                             AuthResourceRepresentation.ofStreamInScope(scopeName, streamName),
-                                             AuthHandler.Permissions.READ))
-                                     .map(m -> StreamInfo.newBuilder().setScope(scopeName).setStream(m).build())
-                                     .collect(Collectors.toList());
-                             return Controller.StreamsInScopeResponse.newBuilder().addAllStreams(streams)
-                                     .setContinuationToken(Controller.ContinuationToken.newBuilder().setToken(
-                                             response.getValue()).build()).build();
-                        }),
+                delegationToken -> Futures.exceptionallyExpecting(controllerService.listStreams(
+                        scopeName, request.getContinuationToken().getToken(), listStreamsInScopeLimit),
+                        e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, null)
+                                          .thenApply(response -> {
+                                              if (response == null) {
```

Comment:
Is it possible that `response` is `null` and not because of a data not found exception? What if there is no stream in the scope, what's the response?

Reply:
No, if there are no children we get empty set .